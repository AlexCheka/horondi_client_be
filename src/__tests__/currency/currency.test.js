/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const currencyService = require('../../modules/currency/currency.service')
const currencyResolver = require('../../modules/currency/currency.resolver')
const client = require('../../utils/apolloClient');

require('dotenv').config();


describe('querries', () => {
    test('should recive all currencies like in service and resolver', async () => {
        const res = await client.query({
            query: gql`
                query {       
                    getAllCurrencies {
                        date
                        convertOptions {
                            name
                            exchangeRate
                        }
                    }
                }
            `
        }).then(res => res)

        expect(currencyService.getAllCurrencies()).resolves.toBe(res);
        expect(currencyResolver.currencyQuery.getAllCurrencies()).resolves.toBe(res);
        expect(res.data.getAllCurrencies).toMatchSnapshot()
    })

    test('should recive currencies by id like in service and resolver', async () => {
      const res = await client.query({
          query: gql`
              query {       
                getCurrencyById(id: "5ef3970c0ab5dd42436dd5c7"){
                    date
                    convertOptions {
                        name
                        exchangeRate
                    }
                }
              }
          `
      }).then(res => res)

      expect(currencyService.getCurrencyById("5ef3970c0ab5dd42436dd5c7")).resolves.toBe(res);
      expect(currencyResolver.currencyQuery.getCurrencyById('',{id:"5ef3970c0ab5dd42436dd5c7"})).resolves.toBe(res);
      expect(res.data.getCurrencyById).toMatchSnapshot()
  })
})