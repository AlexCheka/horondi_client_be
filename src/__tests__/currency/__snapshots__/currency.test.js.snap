// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`querries should recive all currencies like in service and resolver 1`] = `
Array [
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.7544,
        "name": "USD",
      },
      Object {
        "exchangeRate": 29.9609,
        "name": "EUR",
      },
    ],
    "date": "1591228800000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.6953,
        "name": "USD",
      },
      Object {
        "exchangeRate": 29.9975,
        "name": "EUR",
      },
    ],
    "date": "1591315200000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.6005,
        "name": "USD",
      },
      Object {
        "exchangeRate": 30.1477,
        "name": "EUR",
      },
    ],
    "date": "1591660800000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.6412,
        "name": "USD",
      },
      Object {
        "exchangeRate": 30.0872,
        "name": "EUR",
      },
    ],
    "date": "1591747200000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.6367,
        "name": "USD",
      },
      Object {
        "exchangeRate": 30.2859,
        "name": "EUR",
      },
    ],
    "date": "1591833600000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.5999,
        "name": "USD",
      },
      Object {
        "exchangeRate": 30.1723,
        "name": "EUR",
      },
    ],
    "date": "1591920000000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.7303,
        "name": "USD",
      },
      Object {
        "exchangeRate": 30.2119,
        "name": "EUR",
      },
    ],
    "date": "1592179200000",
  },
  Object {
    "convertOptions": Array [
      Object {
        "exchangeRate": 26.844,
        "name": "USD",
      },
      Object {
        "exchangeRate": 30.2062,
        "name": "EUR",
      },
    ],
    "date": "1592265600000",
  },
]
`;

exports[`querries should recive currencies by id like in service and resolver 1`] = `
Object {
  "convertOptions": Array [
    Object {
      "exchangeRate": 26.7544,
      "name": "USD",
    },
    Object {
      "exchangeRate": 29.9609,
      "name": "EUR",
    },
  ],
  "date": "1591228800000",
}
`;
